apps/finance/app/src/components/useFilteredTransfers.js

```diff
@@ -60,19 +60,35 @@ function useFilteredTransfers({ transactions, tokens }) {
         ) {
           return false
         }
-
         // Exclude by date range
+        // Case 1: we have both a start and end date.
         if (
-          // We're not checking for an end date because we will always
-          // have a start date for defining a range.
           selectedDateRange.start &&
+          selectedDateRange.end &&
           !isWithinInterval(new Date(date), {
             start: startOfDay(selectedDateRange.start),
             end: endOfDay(selectedDateRange.end),
           })
         ) {
           return false
         }
+
+        // Case 2: we have a start date, but not an end date.
+        if (
+          selectedDateRange.start &&
+          !selectedDateRange.end &&
+          !isEqual(new Date(date), selectedDateRange.start)
+        ) {
+          return false
+        }
+
+        // Case 3: we don't have a start date, but do have an end date.
```

Comment:
I'm not sure I get these changes...

What I might do instead is filter separately by start date and end date, with `isAfter()` and `isBefore()`.

E.g.

```js
if (selectedDateRange.start && isBefore(new Date(date), startOfDay(selectedDateRange.start)) {
  return false
}
// and vice-versa for the end date
```